src/main/java/de/tum/in/www1/artemis/service/connectors/RemoteArtemisServiceConnector.java

```diff
@@ -123,4 +123,16 @@ static HttpHeaders authenticationHeaderForSecret(String secret) {
         headers.setBearerAuth(secret);
         return headers;
     }
+
+    /**
+     * Helper to generate HttpHeaders for a symmetric secret.
+     *
+     * @param secret Authorization secret
+     * @return HttpHeaders
+     */
+    static HttpHeaders authorizationHeaderForSymmetricSecret(String secret) {
+        HttpHeaders headers = new HttpHeaders();
```

Comment:
We use RestTemplates for all other external connectors. Would this be possible here as well?

Reply:
We actually use a RestTemplate here with this special set of Headers

Reply:
Yes, but as far as I can see, you do not inject it and you do not use an Interceptor here.
Have a look how this is done in `BambooService`, `BambooAuthorizationInterceptor` and in `RestTemplateConfiguration`.

I would prefer that we use a consistent approach for the communication with external systems.